tests/compile_tests.rs

```diff
@@ -0,0 +1,14 @@
+#[test]
+fn runner() {
+    no_std_test();
+}
+
+#[rustversion::nightly]
+fn no_std_test() {
```

Comment:
Check out https://github.com/serde-rs/json/blob/265f1cae423be35b4e02056cae2bbc2afa14ccf4/tests/compiletest.rs#L1 for how I recommend doing this. That makes the test appear as ignored on non-nightly rather than silently passing.